lib/custom_a11y_rules.js

```diff
@@ -156,6 +171,26 @@ var customRules = {
             evaluate: function(node, options) {
                 return axe.utils.isHidden(node);
             },
-        }
+        },
+        {
+            'id': 'section-heading-first-child',
+            'metadata': {
+                'impact': 'minor',
+                'messages': {
+                    'pass': '<section> elements should have a heading as its first child',
+                    'fail': '<section> elements should have a heading as its first child'
+                }
+            },
+            evaluate: function(node, options) {
+                var childNode = node.children[0].nodeName,
+                    validHeadings = ['H1', 'H2', 'H3', 'H4', 'H5', 'H6'];
+                
+                if (validHeadings.indexOf(childNode) > -1) {
```

Comment:
If I'm reading this right, we could have 
```
<section>
  <p>
  <h1>
```
and it would pass. Is that an accurate conclusion? I don't think that's the intention...